src/zocket/Program.cs

```diff
@@ -24,107 +24,96 @@ static async Task<int> Main(string[] args)
                     "--port",
                     getDefaultValue: () => 9999,
                     description: "Port to bind to"),
+                new Option<int>(
+                    "--tls-port",
+                    getDefaultValue: () => 10000),
                 new Argument<string>(
                     "command",
                     getDefaultValue: () => "dotnet watch run",
                     description: "The command to execute with zocket"
                     )
             };
             rootCommand.Description = "zocket";
-            rootCommand.Handler = CommandHandler.Create<int, string>(Listen);
+            rootCommand.Handler = CommandHandler.Create<int, int, string>(Listen);
 
             return await rootCommand.InvokeAsync(args);
         }
 
-        private static Task Listen(int port, string command)
+        private static async Task Listen(int port, int tlsPort, string command)
         {
-            if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
-            {
-                return ListenWindows(port, command);
-            }
-            else if (RuntimeInformation.IsOSPlatform(OSPlatform.Linux))
-            {
-                return ListenLinux(port, command);
-            }
-            return Task.CompletedTask;
-        }
-
-        private static Task ListenLinux(int port, string command)
-        {
-            var exitEvent = new ManualResetEvent(false);
+            using var cts = new CancellationTokenSource();
             Console.CancelKeyPress += delegate (object sender, ConsoleCancelEventArgs e)
             {
                 e.Cancel = true;
-                exitEvent.Set();
+                cts.Cancel();
             };
-            var ipEndPoint = new IPEndPoint(IPAddress.Loopback, port);
-            using var listenSocket = new Socket(ipEndPoint.AddressFamily,
-                                                SocketType.Stream,
-                                                ProtocolType.Tcp);
-            listenSocket.Bind(ipEndPoint);
-            var duplicatedSocket = listenSocket.DuplicateSocketLinux();
+
             var parsedCommand = command.Split(' ', 2);
             ProcessStartInfo psi = parsedCommand.Length switch
             {
                 1 => new ProcessStartInfo(parsedCommand[0]),
                 2 => new ProcessStartInfo(parsedCommand[0], parsedCommand[1]),
-                _ => throw new ArgumentException(),
+                _ => default
             };
 
+            using var httpListenSocket = SetupSocket(port);
+            using var httpsListenSocket = SetupSocket(tlsPort);

```

Comment:
For the future: consider adding an option to pass in the path to a launchsettings.json, and we parse each URL out of it. This is specifically important because by default dotnet watch doesn't look at launch settings.

Reply:
I'll do it in a separate PR. It's quite a big change